src/core/utilities/types.hpp

```diff
@@ -241,11 +244,20 @@ namespace core
 namespace PHARE::core
 {
 template<typename Container, typename Multiplies = typename Container::value_type>
-Multiplies product(Container const& container, Multiplies mul = 1)
+constexpr Multiplies product(Container const& container, Multiplies init = 1)
```

Comment:
I think the point of this `Multiplies` type is to be able to override the return type

https://github.com/PhilipDeegan/PHARE/blob/mkn/src/core/utilities/types.hpp#L254

Reply:
and the version I pushed breaks that?

Reply:
no, but I don't get the copy from `init` to `result`

Reply:
´Cause you’re too smart